### ISSUE TITLE
[lldb] SBValue::GetAsUnsigned overflow doesn't respect underlying integer type range

### ISSUE THREAD

**Issue by Michael137** [llvm/llvm-project#58032]
Based on discussion in [D134493](https://reviews.llvm.org/D134493#3811225).

Repro:
```
❯ cat overflow.cc 
#include <stdint.h>
int main() {
  int8_t i8 = -1;
  int32_t i32 = -1;
  int64_t i64 = -1;
  return 0;
}

(lldb) script

>>> lldb.frame.FindVariable("i8").GetValueAsUnsigned()
4294967295
>>> lldb.frame.FindVariable("i32").GetValueAsUnsigned()
4294967295
>>> lldb.frame.FindVariable("i64").GetValueAsUnsigned()
18446744073709551615
>>>
```

Observe how the `i8` overflow is actually a u32 instead of the `255` which is what one would expect based on the behaviour of the last two overflows. Probably a bug we should fix if feasible

**Comment by Michael137** [comment]
@werat FYI

**Comment by werat** [comment]
> Observe how the `i8` overflow is actually a `u32` instead of the `255` which is what one would expect based on the behaviour of the last two overflows. 

Hm, I guess this depends on what the user understands by "get value as unsigned". My understanding was that means "interpret the value as `uint64_t`" (which is the return value of the function). So I would expect all three to return `18446744073709551615`.

@labath @shafik what do you think the this should do?

**Comment by Michael137** [comment]
> > Observe how the `i8` overflow is actually a `u32` instead of the `255` which is what one would expect based on the behaviour of the last two overflows.
> 
> Hm, I guess this depends on what the user understands by "get value as unsigned". My understanding was this means "interpret the value as `uint64_t`" (which is the return value of the function). So I would expect all three to return `18446744073709551615`.
> 
> @labath @shafik what do you think the this should do?

Ah yes that does look like more appropriate behaviour. Misunderstood

**Comment by labath** [comment]
Yes, I think that would be the least surprising behaviour, although I don't think we can implement this in a way that wouldn't surprise someone.

If we could change the names of these things, then I'd say we should change them to `Get(Un)signExtendedValue`, which would make it clear what is happening (and that's actually how llvm::AP(S)Int names these things).
